docs/modules/ROOT/pages/references/functional-requirements/billing.adoc

```diff
@@ -0,0 +1,172 @@
+= Billing and Reporting
+
+NOTE: This page is currently very VSHN specific and targets the https://products.docs.vshn.ch/products/appuio/cloud/pricing.html[APPUiO Cloud price model].
+
+Each organization gets an invoice, containing all items of the organization in a summarized format.
+Detailed information can be looked up in the {portal}.
+
+== Features
+
+A list of features the system should have:
+
+* Versioned price lists, including a default price list
+* Custom price plan per organization
+* Discounting
+* Pre-Payment / Credit Management / Credit Vouchers
+* Invoice payment by credit-card (Automated / Manually)
+* Access to current running costs (Dashboard)
+* Access to invoices (PDF)
+
+== Example Invoice
+
+.Example invoice for APPUiO Cloud
+====
+_Invoice period: 1.1. - 31.1. (44'640 minutes, timezone UTC)._
+
+[cols="4,1,1,1,1,1", options="header", stripes="hover"]
+|===
+|Description
+|Quantity
+|Unit of Measure
+|Unit Price
+|Discount
+|Total (Rounded)
+
+6+e|*Compute*
+
```

Comment:
I looked at the preview and can't help but wonder if a sorting by namespace makes more sense than sort by product category.

The current example lists all compute resources by zone and namespace.
Then it lists all storage resources by zone and namespace.
Then maybe LoadBalancer, or other categories, by zone and namespace and so on.

This makes it hard to follow what exactly a namespace is costing me, as I'd have to search and sum up resources from all over the place.

What if instead the line items are combined and summarized by namespace instead?

For example (just prosa ASCII art):
```
                                  Quantity | Unit  | ...
========================================================
zone: cloudscale.ch LPG 2
namespace: my-app-prod

Compute
--------------------------------------------------------
Usage (Min / Max / Avg): 
128 MiB / 1034 MiB / 585.66 MiB  |  7028   |  MiB  |  ...
--------------------------------------------------------

Persistent Storage (PVC)
--------------------------------------------------------
Usage (Avg): 
10 GiB                           |  100    |  GiB  |  ...
--------------------------------------------------------

Total for namespace
--------------------------------------------------------
                                                    0.02

========================================================
zone: Exoscale GVA 1
namespace: my-app-test

Compute
--------------------------------------------------------
Usage (Min / Max / Avg): 
...
```

Ofc, not all namespaces have storage. and some may have storage without compute. In those cases the listing for a category for each namespace just disappears. The "Total for namespace" then basically gives summarizes the cost for a namespace.

A decision-maker could then better decide where to optimize cost for applications. e.g it allows to say "this test app costs us x amount per month. Do we need it?"

Reply:
I second that idea.